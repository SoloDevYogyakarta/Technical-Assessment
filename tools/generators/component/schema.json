{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "component",
  "type": "object",
  "properties": {
    "filename": {
      "type": "string",
      "description": "Write your file name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your filename"
      },
      "$default": {
        "$source": "filename"
      }
    },
    "directoryname": {
      "type": "string",
      "description": "Write your directory name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your directoryname"
      },
      "$default": {
        "$source": "directoryname"
      }
    },
    "projectname": {
      "type": "string",
      "description": "Write your project name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your projectname"
      },
      "$default": {
        "$source": "projectname"
      }
    }
  }
}
