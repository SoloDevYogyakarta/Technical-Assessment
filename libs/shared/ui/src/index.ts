export * from './lib/shared-ui.module';
export * from './lib/services/notification-service/notification-service';
