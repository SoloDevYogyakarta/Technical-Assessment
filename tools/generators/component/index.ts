import {
  Tree,
  formatFiles,
  getProjects,
  names,
  generateFiles,
} from '@nrwl/devkit';
import { join } from 'path';
import { ISchema } from '../schema';

export default async function (tree: Tree, schema: ISchema) {
  const workspace = getProjects(tree).get(schema.projectname);
  const templatePath = 'tools/generators/component/templates';
  const directoryPath = schema.directoryname
    ? `${workspace?.sourceRoot}/lib/${schema.directoryname}`
    : `${workspace?.sourceRoot}/lib`;
  const filename = `${schema.filename}.component`;
  const targetPath = join(directoryPath, filename);
  const componentNames = names(filename);
  const substitutions = {
    tmpl: '',
    ...componentNames,
  };
  generateFiles(tree, templatePath, targetPath, substitutions);
  await formatFiles(tree);
}
