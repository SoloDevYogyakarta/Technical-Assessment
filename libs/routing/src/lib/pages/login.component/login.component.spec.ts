import { LoginComponent } from './login.component';
import {
  TestBed,
  TestBedStatic,
  ComponentFixture,
} from '@angular/core/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { NotificationService } from '@nov9-technical-assessment/shared/ui';

describe('LoginComponent', () => {
  let container: TestBedStatic;
  let component: ComponentFixture<LoginComponent>;

  beforeEach(async () => {
    container = await TestBed.configureTestingModule({
      imports: [FormsModule, ReactiveFormsModule],
      declarations: [LoginComponent],
      providers: [NotificationService],
    }).compileComponents();

    component = TestBed.createComponent(LoginComponent);
    component.detectChanges();
  });

  it('should render snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it("should have as title 'LoginComponent'", () => {
    const template = component.componentInstance;
    expect(template.title).toEqual('LoginComponent');
  });

  it('should render title', () => {
    const template = component.nativeElement as HTMLElement;
    expect(template.querySelector('form')?.textContent).toContain(
      'Log in now !'
    );
  });

  afterEach(async () => {
    await jest.clearAllMocks();
  });
});
