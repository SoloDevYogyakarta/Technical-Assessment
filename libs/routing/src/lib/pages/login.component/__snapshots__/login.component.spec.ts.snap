// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should render snapshot 1`] = `
<login-component
  __ngContext__="0"
  form={[Function FormGroup]}
  router={[Function Router]}
  title={[Function String]}
>
  <form
    class="w-full flex items-center justify-center flex-col ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <h1
      class="text-gray-600 text-2xl font-semibold py-16 capitalize"
    >
       Log in now ! 
    </h1>
    <div
      class="w-full max-w-md"
    >
      <label
        class="font-sans font-semibold text-sm text-gray-500"
        for="email"
      >
        Email Address
      </label>
      <input
        class="w-full outline-none border-[1px] border-gray-300 rounded-lg py-2 transition duration-500 hover:border-green-600 px-3 text-gray-500 mb-3 ng-untouched ng-pristine ng-valid"
        formcontrolname="email"
        id="email"
        name="email"
        placeholder="Email Address"
        type="text"
      />
      <label
        class="font-sans font-semibold text-sm text-gray-500"
        for="password"
      >
        Password
      </label>
      <input
        class="w-full outline-none border-[1px] border-gray-300 rounded-lg py-2 transition duration-500 hover:border-green-600 px-3 text-gray-500 mb-3 ng-untouched ng-pristine ng-valid"
        formcontrolname="password"
        id="password"
        name="password"
        placeholder="Password"
        type="password"
      />
      <div
        class="flex items-center flex-col justify-center"
      >
        <button
          class="w-full py-2 rounded-lg bg-material text-white"
          type="submit"
        >
           Sign in 
        </button>
        <a
          class="text-center font-sans text-sm text-gray-600 transition duration-500 cursor-pointer hover:text-blue-400 mt-4"
        >
           You don't have an account? 
        </a>
      </div>
    </div>
  </form>
</login-component>
`;
